### ISSUE TITLE
Doesn't redirect from Pinterest Application

### ISSUE THREAD

**Issue by dzbinomial** [pinterest/ios-pdk#123]
iOS 9.*, 10.*
Not working for Pinterest app versions 6.1, 6.11, 6.12, 6.13
Works fine for Pinterest app version 6.0.4
pod : PinterestSDK 1.0.2

Pin created from the iOS application (which is registered with my Pinterest developer account), iOS application redirects to Pinterest application which opens "Choose board" screen.
After successful pin or canceled by user action Pinterest app does not redirects to my iOS app.

Here is the "pinitURL" from "PDKPin.m" file.
-------------------------------------------
pinterestsdk.v1://pinit/?source_url=http%253A%252F%252Fwww.news.am&app_name=YeeeeyPin&image_url=http%253A%252F%252Fwww.happydogheaven.com%252Fwp-content%252Fuploads%252Fjet-black-siberian-husky-israel-stunning-siberian-husky-puppies.jpg&app_id=4806047872736839051&suggested_board_name=&description=Description%2520for
-------------------------------------------

**Comment by arunranga** [comment]
@dzbinomial : does any error get displayed?

**Comment by dzbinomial** [comment]
Nope, it even successfully pins and displays "Saved to {board name}".

**Comment by rahul-malik** [comment]
@dzbinomial - Can you try with the latest version of the Pinterest client (6.15) and report back if this is still happening? There is an alternative approach as well where you can create the Pin from your application by handling authentication.

Docs: https://developers.pinterest.com/docs/sdks/ios/
>If you want the user to save Pins without leaving your app, you need to authenticate the user with Pinterest using authenticateWithPermissions:withSuccess:andFailure: on PDKClient.

**Comment by mstratis** [comment]
@rahul-malik This is still an issue for me while using the latest Pinterest client (6.15) when my application launches into the Pinterest client for authorization while the user was not previously signed in to the client. Upon successful sign in, the Pinterest client dismisses the sign in modal and never presents the prompt for the user to allow permissions to my application resulting in no redirect back to the application requesting the oauth token. However, if the user was already signed in to the Pinterest client prior to performing the oauth request from my application, the functionality works as expected (with the "this application would like your permission" screen displaying and redirecting back to my application after the user makes a decision.)

**Comment by rahul-malik** [comment]
@mstratis - Thank you for the clear description and steps to reproduce. I'll follow up internally to see what the issue might be but can't guarantee any quick resolution within the Pinterest application with the App Store shutting down for the holidays soon.

**Comment by larryonoff** [comment]
Any updates on this issue?

**Comment by rahul-malik** [comment]
@larryonoff thanks for the ping. I'll follow up internally ok the status of that bug but in the meanwhile I'd recommend the alternative approach I mentioned earlier

**Comment by larryonoff** [comment]
@rahul-malik we weren't able make it works with approach you mentioned earlier.
